spec/support/mock_auth_api_helper.rb

```diff
@@ -0,0 +1,54 @@
+module MockAuthApiHelper
+  MockAuthApiNotSetup = Class.new(Exception)
+
+  def mock_auth_api_client(setup)
+    @old_service = ServiceRegistry.service(:auth_api_client)
```

Comment:
We shouldn't need to keep a reference to the registered service as we should be able to just assign the mock to the registered key. Teardown is implicit as it gets set to the actual client in the initializer.